### PR TITLE
Move APIClient into classy-pay-core and refactor it

### PR DESCRIPTION
**Pull request by brownjava**
Cleans the code up a bit and adds types.

### CODE REVIEW

File: src/APIClient.ts [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+   * Builds an API client
+   *
+   * @param {string} clientId your client ID
+   * @param {string} clientSecret your client secret (password)
+   * @param {string} oauthUrl oauth URL
+   * @param {string} apiUrl api URL
+   * @param {number} timeout HTTP request timeout, or 120 minutes if unspecified
+   */
+  constructor(clientId: string, clientSecret: string, oauthUrl: string, apiUrl: string, timeout: number = 120000) {
+    if (!clientId) throw Error('Cannot construct APIClient with null clientId');
```
[the remark this answers is not in the document]
Comment (timothyk7) [reply src/APIClient.ts]: for backwards compatibility - JB

File: src/utils/utils.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+export const safeStrConcat = (strings: Array<any>): string => {
```
[the remark this answers is not in the document]
Comment (timothyk7) [reply src/utils/utils.ts]: for backwards compatibility - JB

File: src/APIClient.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (brownjava) [reply src/APIClient.ts]: Right.  This is caught at compile time with TS, but with JS we want to bail if these aren't provided.

File: src/utils/utils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (brownjava) [reply src/utils/utils.ts]: Yeah, I think the intent here is we want to be able to take any array containing any type, and concat it into one string ignoring null and undefined values.

**Pull request merged**